### PR TITLE
fix: send throttling rate limit per second

### PR DESCRIPTION
**Pull request by BonapartePC**
# Description

Previously, we are only sending throttling rate limit as a metric without a time window. This PR converts the rate limit to per second.

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: router/throttler/factory_test.go [hunk 1]
```diff
@@ -71,6 +74,27 @@ func TestFactory(t *testing.T) {
...
+		config := config.New()
+		config.Set("Router.throttler.adaptive.enabled", true)
+		maxLimit := int64(300)
+		window := 2 * time.Second
+		config.Set("Router.throttler.adaptive.maxLimit", maxLimit)
+		config.Set("Router.throttler.adaptive.minLimit", int64(100))
+		config.Set("Router.throttler.adaptive.timeWindow", window)
+		ctrl := gomock.NewController(t)
+		mockStats := mock_stats.NewMockStats(ctrl)
+		mockMeasurement := mock_stats.NewMockMeasurement(ctrl)
```
Comment (lvrach) [inline router/throttler/factory_test.go:88]: could [memstat](https://github.com/rudderlabs/rudder-go-kit/tree/main/stats/memstats) be more suitable than mock_stats?

**Review by lvrach — APPROVED** [review]

File: router/throttler/factory_test.go [hunk 2]
```diff
@@ -71,6 +74,27 @@ func TestFactory(t *testing.T) {
...
+		mockStats.EXPECT().NewTaggedStat("throttling_rate_limit", stats.GaugeType, gomock.Any()).Times(1).Return(mockMeasurement)
+		mockStats.EXPECT().NewTaggedStat("adaptive_throttler_limit_factor", stats.GaugeType, gomock.Any()).Times(1).Return(mockMeasurement)
+		mockMeasurement.EXPECT().Gauge(maxLimit / getWindowInSecs(window)).Times(1)
+		mockMeasurement.EXPECT().Gauge(float64(1)).AnyTimes()
+		f, err := NewFactory(config, mockStats)
+		require.NoError(t, err)
+		defer f.Shutdown()
+		f.Get("destName", "destID")
```
Comment (achettyiitr) [inline router/throttler/factory_test.go:96]: We can explicitly declare that we ignore it?
```suggestion
		_ = f.Get("destName", "destID")
```

File: router/throttler/factory_test.go (hunk 1 above)
Comment (achettyiitr) [reply router/throttler/factory_test.go:88]: +1 for mem stats. One downside with the current mocks is we are not verifying the tags and well as which Gauge metrics is emitting what? With memstats, we would need to do it?

```GO
	t.Run("check stats when adaptive is enabled", func(t *testing.T) {
		maxLimit := int64(300)
		window := 2 * time.Second

		c := config.New()
		c.Set("Router.throttler.adaptive.enabled", true)
		c.Set("Router.throttler.adaptive.maxLimit", maxLimit)
		c.Set("Router.throttler.adaptive.minLimit", int64(100))
		c.Set("Router.throttler.adaptive.timeWindow", window)

		statsStore, err := memstats.New()
		require.NoError(t, err)

		f, err := NewFactory(c, statsStore)
		require.NoError(t, err)
		defer f.Shutdown()

		_ = f.Get("destName", "destID")

		require.EqualValues(t, maxLimit/getWindowInSecs(window), statsStore.Get("throttling_rate_limit", stats.Tags{
			"destinationId": "destID",
			"destType":      "destName",
			"adaptive":      "true",
		}).LastValue())
		require.EqualValues(t, 1, statsStore.Get("adaptive_throttler_limit_factor", stats.Tags{
			"destinationId": "destID",
			"destType":      "destName",
		}).LastValue())
	})

File: router/throttler/factory_test.go [hunk 3]
```diff
@@ -71,6 +74,27 @@ func TestFactory(t *testing.T) {
 			return ta.getLimit() == int64(0)
 		}, 2*time.Second, 100*time.Millisecond)
 	})
+
+	t.Run("check stats when adaptive is enabled", func(t *testing.T) {
+		config := config.New()
```
Comment (achettyiitr) [inline router/throttler/factory_test.go:79]: `Variable 'config' collides with imported package name`. let's some other variable name in here, pls?
```suggestion
		conf := config.New()
```

**Review by achettyiitr — APPROVED** [review]

File: router/throttler/factory_test.go (hunk 1 above)
Comment (BonapartePC) [reply router/throttler/factory_test.go:88]: Done

File: router/throttler/factory_test.go (hunk 2 above)
Comment (BonapartePC) [reply router/throttler/factory_test.go:96]: Done

File: router/throttler/factory_test.go (hunk 3 above)
Comment (BonapartePC) [reply router/throttler/factory_test.go:79]: Done

**Pull request merged**